vcd/resource_vcd_vapp_vm_test.go

```diff
@@ -152,8 +160,163 @@ func TestAccVcdVAppVm_Clone(t *testing.T) {
 	})
 }
 
-func init() {
-	testingTags["vm"] = "resource_vcd_vapp_vm_test.go"
+// TestAccVcdVappVm_NicIndex aims to replicate issue when VM is created outside of Terraform and
+// NICs are not indexed starting with 0. It is not possible to create a VM with network cards
+// starting with index other than 0 in terraform-provider-vcd however it can happen when VM is
+// imported from other systems.
+//
+// This test creates a VM using go-vcloud-director SDK with such vCD NIC indexes:
+// * vCD NIC index 1 (is_primary=false) - Terraform network block index 0 (is_primary=false)
+// * vCD NIC index 2 (is_primary=true) - Terraform network block index 1 (is_primary=true)
+// Before resolving below mentioned issue terraform-provider-vcd would report is_primary=false for
+// both NICs.
+// GitHub issue: https://github.com/terraform-providers/terraform-provider-vcd/issues/458
+func TestAccVcdVappVm_NicIndex(t *testing.T) {
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	vapp, _, err := testAccVcdVappVmNicIndexcreateVappVm(t.Name()+"-vApp", t.Name())
+	if err != nil {
+		t.Errorf("error creating VM using SDK: %s", err)
+	}
+	// The VM was created therefore it must be deleted afterwards
+	defer func() {
+		if vapp != nil {
+			task, err := vapp.Delete()
+			if err != nil {
+				t.Errorf("error deleting vApp after test: %s", err)
+			}
+			err = task.WaitTaskCompletion()
+			if err != nil {
+				t.Errorf("error waiting for vApp deletion: %s", err)
+			}
+		}
+	}()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// The step below imports an existing VM (created via go-vcloud-director SDK) with odd NIC
```

Comment:
Any chance we can visualize the imported resource when GOVCD_DEBUG is set?

Reply:
Added something like this when GOVCD_DEBUG is enabled:
```
=== RUN   TestAccVcdVappVm_NicIndex
computer_name = VM-001
cpu_cores = 1
cpus = 2
customization.# = 1
customization.0.allow_local_admin_password = true
customization.0.auto_generate_password = true
customization.0.change_sid = false
customization.0.enabled = false
customization.0.force = false
customization.0.join_domain = false
customization.0.join_org_domain = false
customization.0.must_change_password_on_first_login = false
customization.0.number_of_auto_logons = 0
description = created by test
disk.# = 0
expose_hardware_virtualization = false
guest_properties.% = 0
hardware_version = vmx-13
href = https://192.168.1.109/api/vApp/vm-a3e8e99a-dde5-4761-a51b-e9feee092fad
internal_disk.# = 1
internal_disk.0.bus_number = 0
internal_disk.0.bus_type = paravirtual
internal_disk.0.disk_id = 2000
internal_disk.0.iops = 0
internal_disk.0.size_in_mb = 16384
internal_disk.0.storage_profile = *
internal_disk.0.thin_provisioned = true
internal_disk.0.unit_number = 0
memory = 512
metadata.% = 0
name = TestAccVcdVappVm_NicIndex
network.# = 2
network.0.adapter_type = VMXNET3
network.0.ip_allocation_mode = NONE
network.0.is_primary = false
network.0.mac = 00:50:56:01:43:a9
network.0.type = none
network.1.adapter_type = VMXNET3
network.1.ip_allocation_mode = NONE
network.1.is_primary = true
network.1.mac = 00:50:56:01:43:aa
network.1.type = none
org = my-org
os_type = debian10Guest
override_template_disk.# = 0
storage_profile = *
vapp_name = TestAccVcdVappVm_NicIndex-vApp
vdc = my-org-vdc2

--- PASS: TestAccVcdVappVm_NicIndex (27.52s)
```